open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +257,427 @@ public static JSONArray appendDummyAccountID() {
+            controlParameterArray.add(formattedAmount);
+            jsonElementControlParameter.appendField(ConsentExtensionConstants.DATA_SIMPLE, controlParameterArray);
+
+            consentDataJSON.add(jsonElementControlParameter);
+        }
+
+            // Constructing jsonElementPeriodAlignment
+            JSONObject jsonElementPeriodAlignment = new JSONObject();
+            jsonElementPeriodAlignment.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.CONTROL_PARAMETER_PERIOD_ALIGNMENT_TITLE);
+            String periodAlignment = (String) ((JSONObject) ((JSONArray) controlParameters
+                    .get(ConsentExtensionConstants.PERIODIC_LIMITS)).get(0))
+                    .get(ConsentExtensionConstants.PERIOD_ALIGNMENT);
+
+        if (periodAlignment != null) {
+            JSONArray periodAlignmentArray = new JSONArray();
+            periodAlignmentArray.add(periodAlignment);
+            jsonElementPeriodAlignment.appendField(ConsentExtensionConstants.DATA_SIMPLE, periodAlignmentArray);
+            consentDataJSON.add(jsonElementPeriodAlignment);
+        }
+
+            // Constructing jsonElementPeriodType
+            JSONObject jsonElementPeriodType = new JSONObject();
+            jsonElementPeriodType.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.CONTROL_PARAMETER_PERIOD_TYPE_TITLE);
+            String periodType = (String) ((JSONObject) ((JSONArray) controlParameters
+                    .get(ConsentExtensionConstants.PERIODIC_LIMITS)).get(0)).get(ConsentExtensionConstants.PERIOD_TYPE);
+
+        if (periodType != null) {
+            JSONArray periodTypeArray = new JSONArray();
+            periodTypeArray.add(periodType);
+            jsonElementPeriodType.appendField(ConsentExtensionConstants.DATA_SIMPLE, periodTypeArray);
+            consentDataJSON.add(jsonElementPeriodType);
+        }
+
+            // Constructing jsonElementPeriodicLimits
+            JSONObject jsonElementPeriodicLimits = new JSONObject();
+            jsonElementPeriodicLimits.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.CONTROL_PARAMETER_AMOUNT_TITLE + periodType);
+            JSONArray periodicLimitsArray = new JSONArray();
+
+            JSONObject periodicLimitsObject = (JSONObject) ((JSONArray) controlParameters
+                    .get(ConsentExtensionConstants.PERIODIC_LIMITS)).get(0);
+
+        if (periodicLimitsObject != null) {
+            String currency = (String) periodicLimitsObject.get(ConsentExtensionConstants.CURRENCY);
+            String amount = (String) periodicLimitsObject.get(ConsentExtensionConstants.AMOUNT);
+            String formattedPeriodicAmount = String.format("%s %s", currency, amount);
+
+            periodicLimitsArray.add(formattedPeriodicAmount);
+
+            jsonElementPeriodicLimits.appendField(ConsentExtensionConstants.DATA_SIMPLE, periodicLimitsArray);
```

Comment:
Here also we are appending `DATA_SIMPLE` as in L654, L640 right?

Reply:
Fixed in [fa29ae7]

Reply:
Fixed in [fa29ae7]